### PR TITLE
Added legacyTar option

### PR DESCRIPTION
**Pull request by Xflofoxx**
Added legacy tar option for opening Pre-POSIX.1-1988 (i.e. v7) tar.
See issue https://github.com/kevva/decompress/issues/49

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,40 +1,109 @@
 {
-  "name": "decompress-tar",
-  "version": "4.1.0",
-  "description": "decompress tar plugin",
-  "license": "MIT",
-  "repository": "kevva/decompress-tar",
+  "_args": [
```
Comment (SamVerschueren) [inline package.json]: Revert `package.json`. Don't start from the `npm install` file.

File: index.js [hunk 2]
```diff
@@ -3,53 +3,53 @@ const fileType = require('file-type');
...
+    extract.then = promise.then.bind(promise);
+    extract.catch = promise.catch.bind(promise);
+
+    if (Buffer.isBuffer(input)) {
+        extract.end(input);
+    } else {
+        input.pipe(extract);
+    }
+
+    return extract;
```
Comment (SamVerschueren) [inline index.js]: Don't change indentation. Use tabs.

File: index.js [hunk 3]
```diff
@@ -3,53 +3,53 @@ const fileType = require('file-type');
...
-
-	if (Buffer.isBuffer(input)) {
-		extract.end(input);
-	} else {
-		input.pipe(extract);
-	}
-
-	return extract;
+module.exports = () => (input, opts) => {
+    if (!Buffer.isBuffer(input) && !isStream(input)) {
```
Comment (SamVerschueren) [inline index.js]:
```js
opts = Object.assign({
    legacyTar: false
}, opts);
```

This makes the options object optional and sets a default value for `legacyTar`.

File: readme.md [hunk 4]
```diff
@@ -28,7 +28,7 @@ decompress('unicorn.tar', 'dist', {
 
 ## API
 
-### decompressTar()(input)
+### decompressTar()(input, options)
```
Comment (SamVerschueren) [inline readme.md]: `decompressTar()(input, [options])`

File: readme.md [hunk 5]
```diff
@@ -38,6 +38,14 @@ Type: `Buffer` `Stream`
...
 Buffer or stream to decompress.
 
+#### options
+
+##### legacyTar
+
+Type: `boolean`<br>
+Default: `false`
+
+Prevent file check for [pre-POSIX.1-1988 (i.e. v7)](https://en.wikipedia.org/wiki/Tar_(computing)#File_format) tar files where there is no magic number for recognition. Should use if is generated from old Linux releases.
```
Comment (SamVerschueren) [inline readme.md]: Add an extra newline below this one.

**Comment by sindresorhus** [comment]
Per https://github.com/sindresorhus/file-type/issues/76#issuecomment-277231742, it should be possible to detect it in `file-type`, so I'm gonna pass on this. Every new option adds overhead to the code and the users, and this one feels too obscure.

**Pull request closed without merging**